modules/ting_subsearch_opensuggestion/ting_subsearch_opensuggestion.module

```diff
@@ -0,0 +1,175 @@
+      if ($secondary_result->numTotalObjects > 0) {
+        $params['query'] = array(
+          'original-search' => $search_result->search_key,
+        );
+        drupal_goto('search/ting/' . strtolower($suggestion), $params);
+      }
+    }
+  }
+
+  if ($results->numTotalObjects <= variable_get('ting_subsearch_os_trigger_limit', 5)) {
+    $suggestion = ting_subsearch_opensuggestion_suggest_keys($keys);
+    if ($suggestion) {
+      // Do additional search.
+      $suggested_result = ting_subsearch_common_do_secondary_search($suggestion);
+      if ($suggested_result->numTotalObjects >= variable_get('ting_subsearch_os_message_limit', 10)) {
+
+        // Set message!
+        $msg = t('The search for "!keys" returned !num-results hits. Did you mean "<a href="/search/ting/!suggested-keys">!suggested-keys</a>?" which will return !suggested-num-results hits.', array(
+          '!suggested-keys' => strtolower($suggestion),
+          '!suggested-num-results' => $suggested_result->numTotalObjects,
+          '!keys' => $keys,
+          '!num-results' => $results->numTotalObjects,
+        ));
+        $messages[] = '<div class="messages warning">' . $msg . '</div>';
+      }
+    }
+  }
+  return $messages;
+}
+
+/**
+ * Find the first suggestion for given keys.
+ *
+ * @param string $keys
+ *   The original keys that you want to find suggestion for.
+ *
+ * @return string
+ *   The suggestion.
+ */
+function ting_subsearch_opensuggestion_suggest_keys($keys) {
+
+  if (!variable_get('ting_search_autocomplete_suggestion_url', FALSE)) {
+    return FALSE;
+  }
+
+  $settings = ting_search_autocomplete_settings();
+  $url = variable_get('ting_search_autocomplete_suggestion_url', FALSE) . 'rest/facetSpell';
+
+  $params = $options['query'] = array('query' => check_plain($keys)) + $settings;
+  $options['maxTime'] = 300;
+
+  try {
+    $client = new HttpClient();
+    $response = $client->get(
+      $url,
+      array(
+        'query' => $params,
+        'headers' => array(
+          'Accept' => 'application/json',
+        ),
+      )
+    );
+    $result = json_decode($response->getBody());
+  }
+  catch (GuzzleClientException $e) {
+    throw new Exception($e->getMessage(), $e->getCode(), $e);
+  }
+
+  $items = array();
+  // First child of $result changes name from suggestions to suggestion when
+  // switching between version 3.0 and 3.5.
+  if (!empty(current($result))) {
+    foreach (current($result) as $suggestion) {
+      return $suggestion->phrase;
+    }
+  }
+  if (empty($items)) {
+    return FALSE;
+  }
+}
+
+
+/**
+ * Implements hook_libraries_info().
+ */
+function ting_subsearch_opensuggestion_libraries_info() {
+  return array(
+    'guzzle' => array(
+      'name' => 'Guzzle',
+      'vendor url' => 'https://github.com/guzzle/guzzle',
+      'download url' => 'https://github.com/guzzle/guzzle',
+      'version' => '6.2',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
```

Comment:
The issues reported by Scrutinizer regarding indentation isen't in my opinion an issue.
Locally PHP_CodeSniffer reports this as OK. Perhaps Scrutinizer is running an older version of PHP_CodeSniffer ?